### ISSUE TITLE
compileSync of undefined

### ISSUE THREAD

**Issue by jasonLaster** [cypress-io/cypress#185]
On CI, I occasionally get this error:

```
TypeError: Cannot read property 'compileSync' of undefined
  at Snockets.module.exports.Snockets.compileFile (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/snockets/lib/snockets.js:418:37)
  at Snockets.<anonymous> (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/snockets/lib/snockets.js:101:22)
  at /home/ubuntu/.cypress/Cypress/resources/app/node_modules/snockets/lib/snockets.js:110:12
  at HoldingQueue.onComplete (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/snockets/lib/snockets.js:228:18)
  at HoldingQueue.finalize (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/snockets/lib/snockets.js:494:21)
  at /home/ubuntu/.cypress/Cypress/resources/app/node_modules/snockets/lib/snockets.js:307:18
  at /home/ubuntu/.cypress/Cypress/resources/app/node_modules/snockets/lib/snockets.js:403:20
  at Snockets.module.exports.Snockets.stat (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/snockets/lib/snockets.js:368:18)
  at Snockets.module.exports.Snockets.readFile (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/snockets/lib/snockets.js:377:19)
  at Snockets.module.exports.Snockets.updateDirectives (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/snockets/lib/snockets.js:280:19)
  at Snockets.module.exports.Snockets.getCompiledChain (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/snockets/lib/snockets.js:76:19)
  at Object.module.exports.handle (/home/ubuntu/.cypress/Cypress/resources/app/index.js:1557:31)
  at /home/ubuntu/.cypress/Cypress/resources/app/index.js:4981:19
  at Layer.handle [as handle_request] (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/layer.js:82:5)
  at next (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/route.js:110:13)
  at Route.dispatch (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/route.js:91:3)
  at Layer.handle [as handle_request] (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/layer.js:82:5)
  at /home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:267:22
  at Function.proto.process_params (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:321:12)
  at next (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:261:10)
  at Layer.handle [as handle_request] (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/layer.js:78:12)
  at trim_prefix (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:302:13)
  at /home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:270:7
  at Function.proto.process_params (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:321:12)
  at next (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:261:10)
  at compression (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/compression/index.js:217:5)
  at Layer.handle [as handle_request] (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/layer.js:82:5)
  at trim_prefix (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:302:13)
  at /home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:270:7
  at Function.proto.process_params (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:321:12)
  at next (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:261:10)
  at cookieParser (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/cookie-parser/index.js:56:14)
  at Layer.handle [as handle_request] (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/layer.js:82:5)
  at trim_prefix (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:302:13)
  at /home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:270:7
  at Function.proto.process_params (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:321:12)
  at next (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:261:10)
  at expressInit (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/middleware/init.js:23:5)
  at Layer.handle [as handle_request] (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/layer.js:82:5)
  at trim_prefix (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:302:13)
  at /home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:270:7
  at Function.proto.process_params (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:321:12)
  at next (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:261:10)
  at query (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/middleware/query.js:28:5)
  at Layer.handle [as handle_request] (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/layer.js:82:5)
  at trim_prefix (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:302:13)
  at /home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:270:7
  at Function.proto.process_params (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:321:12)
  at next (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:261:10)
  at Function.proto.handle (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/router/index.js:166:3)
  at EventEmitter.app.handle (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/application.js:170:10)
  at Server.app (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/express/lib/express.js:28:9)
  at Server.<anonymous> (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/engine.io/lib/server.js:434:22)
  at Server.<anonymous> (/home/ubuntu/.cypress/Cypress/resources/app/node_modules/socket.io/lib/index.js:260:16)
  at emitTwo (events.js:100:13)
  at Server.emit (events.js:185:7)
  at HTTPParser.parserOnIncoming [as onIncoming] (_http_server.js:537:12)
  at HTTPParser.parserOnHeadersComplete (_http_common.js:103:23)
```

**Comment by jennifer-mann** [comment]
This is likely due to you having a abnormal file type in your `support` directory like `foo.js~`. Its being picked up as a valid file that cypress is trying to serve and then when it actually serves it it cannot compile it due to the extension not being a valid `.js`.

**Comment by brian-mann** [comment]
This will automatically be fixed when we move away from the current build process to supporting babel.

**Comment by brian-mann** [comment]
This was actually fixed in `0.18.0`.
